File: _bak/cpy0/c/ast.c
#include "ast.h"

#include "lexer.h"
#include "parser.h"

AstNode *parse_source(const char *source, const char *filename) {
    TokenVec tokens;
    lex_source(source, &tokens);
    AstNode *module = parse_tokens(&tokens, filename);
    free_tokens(&tokens);
    return module;
}

File: _bak/cpy0/rust/src/ast.rs
use crate::ast_nodes::AstNode;
use crate::lexer::Lexer;

pub fn parse_source(source: &str, _filename: &str) -> AstNode {
    let mut lexer = Lexer::new(source);
    lexer.scan_tokens();
    let tokens = lexer.get_tokens();
    let mut parser = crate::parser::Parser::new(tokens);
    parser.parse()
}